kin-core/src/main/java/kin/core/PaymentWatcher.java

```diff
@@ -0,0 +1,114 @@
+package kin.core;
+
+
+import static kin.core.Utils.checkNotNull;
+
+import android.support.annotation.NonNull;
+import java.math.BigDecimal;
+import java.util.List;
+import kin.core.ServiceProvider.KinAsset;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Memo;
+import org.stellar.sdk.MemoHash;
+import org.stellar.sdk.Operation;
+import org.stellar.sdk.PaymentOperation;
+import org.stellar.sdk.Server;
+import org.stellar.sdk.requests.EventListener;
+import org.stellar.sdk.responses.TransactionResponse;
+import org.stellar.sdk.sse.ServerSentEvent;
+
+/**
+ * Watch blockchain network for ongoing payments involved specific account.
+ * <p>Use {@link #start(WatcherListener)} to start
+ * watching, new payments will be notified on the input {@link WatcherListener WatcherListener&lt;PaymentInfo&gt;}</p>
+ */
+public class PaymentWatcher {
+
+    private static final String CURSOR_FUTURE_ONLY = "now";
+    private final Server server;
+    private final KinAsset kinAsset;
+    private final KeyPair accountKeyPair;
+    private boolean started;
+    private ServerSentEvent serverSentEvent;
+
+    PaymentWatcher(Server server, Account account, KinAsset kinAsset) {
+        this.server = server;
+        this.kinAsset = kinAsset;
+        accountKeyPair = KeyPair.fromAccountId(account.getAccountId());
+    }
+
+    /**
+     * Start watching for payments, use {@link #stop()} to stop watching.
+     * <p><b>Note:</b> Events will be fired on background thread.</p>
+     *
+     * @param listener listener object for payment events
+     */
+    public void start(@NonNull final WatcherListener<PaymentInfo> listener) {
```

Comment:
do we need to synchronise the start and stop methods? meaning that if start is already called the second start doesn't need to start a new server event it can just add a listener. stop will remove listener and last one stop will also stop the server event (just an idea)

Reply:
I'm not sure multiple listeners is necessary and worth the effort, currently, the second start will raise `IllegalStateException`. stop will not do anything if not started already (maybe need to add to documentation)

Reply:
the documentation need to specify it clearly that currently it only supports one listener at a time